exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -68,5 +79,151 @@ func (prwe *prwExporter) shutdown(context.Context) error {
+		for _, resourceMetric := range resourceMetrics {
+			// TODO: add resource attributes as labels, probably in next PR
+			for _, instrumentationMetrics := range resourceMetric.InstrumentationLibraryMetrics {
+				// TODO: decide if instrumentation library information should be exported as labels
+				for _, metric := range instrumentationMetrics.Metrics {
+					// check for valid type and temporality combination
+					if ok := validateMetrics(metric.MetricDescriptor); !ok {
+						dropped++
+						errs = append(errs, fmt.Errorf("invalid temporality and type combination"))
+						continue
+					}
+					// handle individual metric based on type
+					switch metric.GetMetricDescriptor().GetType() {
+					case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64,
+						otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+						if err := prwe.handleScalarMetric(tsMap, metric); err != nil {
+							dropped++
+							errs = append(errs, err)
+						}
+					}
+				}
+			}
+		}
+
+		if err := prwe.export(ctx, tsMap); err != nil {
+			return pdatautil.MetricCount(md), err
+		}
+
+		if dropped != 0 {
+			return dropped, componenterror.CombineErrors(errs)
+		}
+
+		return 0, nil
+	}
+}
+
+// handleScalarMetric processes data points in a single OTLP scalar metric by adding the each point as a Sample into
+// its corresponding TimeSeries in tsMap.
+// tsMap and metric cannot be nil, and metric must have a non-nil descriptor
+func (prwe *prwExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+
+	mType := metric.MetricDescriptor.Type
+
+	switch mType {
+	// int points
+	case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64:
+		if metric.Int64DataPoints == nil {
+			return fmt.Errorf("nil data point field in metric %v", metric.GetMetricDescriptor().Name)
+		}
+
+		for _, pt := range metric.Int64DataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), prwe.namespace)
+			labels := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value: float64(pt.Value),
+				// convert ns to ms
+				Timestamp: convertTimeStamp(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, labels, metric.GetMetricDescriptor().GetType())
+		}
+		return nil
+
+	// double points
+	case otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+		if metric.DoubleDataPoints == nil {
+			return fmt.Errorf("nil data point field in metric %v", metric.GetMetricDescriptor().Name)
+		}
+		for _, pt := range metric.DoubleDataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), prwe.namespace)
+			labels := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value:     pt.Value,
+				Timestamp: convertTimeStamp(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, labels, metric.GetMetricDescriptor().GetType())
```

Comment:
Same, you seem to ignore the CUMULATIVE part

Reply:
Same as https://github.com/open-telemetry/opentelemetry-collector/pull/1577#discussion_r478611429